hap-py: init 0.3.15

apraga opened the pull request:
New package for genetics : helps compared variants called by a pipeline to a reference.

The only tests not passing are due to a difference in float representation (same value but with
different number of digits after decimal).

For best results (quality and running time), the
authors advice to use RTGtools vcfeval as the
engine. So this package requires  #230394

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

apraga committed:
hap-py: init 0.3.15

New package for genetics : helps compared variants
called by a pipeline to a reference.

The only tests not passing are due to a difference
in float representation (same value but with
different number of digits after decimal).

raspher commented:
Result of `nixpkgs-review pr 283881` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

apraga committed:
hap-py: use rtgtools vcfeval as default core

For best results (quality and running time), the
authors advice to use RTGtools vcfeval as the
engine.

Depends on #230394

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3701

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+{ pkgs, stdenv, fetchFromGitHub, boost, zlib, lib, makeWrapper }:
```
```suggestion
{ stdenv, fetchFromGitHub, boost, zlib, lib, makeWrapper }:
```
Using pkgs here is dirty and breaks cross

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+
+let
+  # Bcftools needs perl
+  runtime = with pkgs; [ bcftools htslib my-python perl samtools ];
```
```suggestion
  runtime = [ bcftools htslib my-python perl samtools ];
```

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+let
+  # Bcftools needs perl
+  runtime = with pkgs; [ bcftools htslib my-python perl samtools ];
+  my-python-packages = p: with p; [
+    bx-python
+    pysam
+    cython
+    pandas
+    psutil
+    nose
+    scipy
+  ];
+  my-python = pkgs.python3.withPackages my-python-packages;
```
```suggestion
  python = pkgs.python3.withPackages (p: with p; [
    bx-python
    pysam
    pandas
    psutil
    scipy
  ]);
```

nose is a test dependency usually and cython a compile time only dep

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+  my-python = pkgs.python3.withPackages my-python-packages;
+in
+stdenv.mkDerivation rec {
+  pname = "hap.py";
+  version  = "0.3.15";
+  src = fetchFromGitHub {
+    owner = "Illumina";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-K8XXhioMGMHw56MKvp0Eo8S6R36JczBzGRaBz035zRQ=";
+  };
```
```suggestion
  pname = "hap.py";
  version  = "0.3.15";

  src = fetchFromGitHub {
    owner = "Illumina";
    repo = "hap.py";
    rev = "v${version}";
    sha256 = "sha256-K8XXhioMGMHw56MKvp0Eo8S6R36JczBzGRaBz035zRQ=";
  };

```

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+  # For cmake : boost lib and includedir are in differernt location
+  BOOST_LIBRARYDIR = "${boost.out}/lib";
+  BOOST_INCLUDEDIR = "${boost.dev}/include";
+  patches = [
+    ./hap-py.patch
+    ./cmake.patch
+  ];
```
```suggestion

  patches = [
    ./hap-py.patch
    ./cmake.patch
  ];

```
Why do we need those patches?

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+    ./hap-py.patch
+    ./cmake.patch
+  ];
+  buildInputs = with pkgs; [
```
```suggestion
  buildInputs = [
```

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+    ./cmake.patch
+  ];
+  buildInputs = with pkgs; [
+    autoconf
+    boost
+    bzip2
+    cmake
+    curl
+    htslib
+    my-python
+    rtg-tools
+    xz
+    zlib
```
autoconf and cmake are definitely nativeBuildInputs.

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+    xz
+    zlib
+  ];
+  nativeBuildInputs = [ pkgs.makeWrapper ];
```
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+    zlib
+  ];
+  nativeBuildInputs = [ pkgs.makeWrapper ];
+  propagatedBuildInputs = runtime;
```
Why propagate all things? Please use wrapper

SuperSandro2000 commented on pkgs/by-name/ha/hap-py/package.nix:
```diff
@@ -0,0 +1,63 @@
+  propagatedBuildInputs = runtime;
+  postFixup = ''
+    wrapProgram $out/bin/hap.py \
+      --set PATH ${lib.makeBinPath runtime}
+       --add-flags "--engine-vcfeval-path=${rtg-tools}/bin/rtg
```
```suggestion
      --set PATH ${lib.makeBinPath runtime} \
      --add-flags "--engine-vcfeval-path=${rtg-tools}/bin/rtg
```

rtg-tools doesn't exist

apraga replied:
rtg-tools is the subject of a new PR that you also reviewed  #23039

apraga commented:
Thanks for your helpful review ! The derivation has been improved to match your suggestions (except the one waiting on 
#23039).

apraga commented:
PR needed for this PR has been merged (thanks !). If you are OK with this PR, could you merge it too ?
